sdk/iot/core/tests/cmocka/az_iot_telemetry_tests.c

```diff
@@ -0,0 +1,283 @@
+#include <az_iot_hub_client.h>
+#include <az_span.h>
+
+#include <stdio.h>
+
+#include <setjmp.h>
```

Comment:
Do we really want to take a dependency on something using setjmp for running tests on 8 and 16bit MCUs?

Reply:
That's a discussion which was briefly had on last Friday which can be brought up separately if you'd like. I think for now we'll keep it.